packages/expo-av/ios/EXAV/AudioSampleCallback/AudioSampleCallbackWrapper.mm

```diff
@@ -0,0 +1,76 @@
+// Copyright 2017-present 650 Industries. All rights reserved.
+
+#import <EXAV/AudioSampleCallbackWrapper.h>
+
+namespace expo {
+namespace av {
+
+std::shared_ptr<LongLivedObjectCollection> AudioSampleCallbackWrapper::callbackCollection =
+  std::make_shared<LongLivedObjectCollection>();
+
+AudioSampleCallbackWrapper::AudioSampleCallbackWrapper(jsi::Function &&callback,
+                           jsi::Runtime &runtime,
+                           std::shared_ptr<CallInvoker> jsInvoker)
+: weakWrapper(JsiCallbackWrapper::createWeak(callbackCollection, std::move(callback), runtime, jsInvoker)) {}
+
+AudioSampleCallbackWrapper::~AudioSampleCallbackWrapper() {
+  auto strongWrapper = weakWrapper.lock();
+  if (strongWrapper) {
+    strongWrapper->destroy();
+  }
+}
+
+void AudioSampleCallbackWrapper::call(AudioBuffer* buffer, double timestamp)
+{
+  auto strongWrapper = this->weakWrapper.lock();
+  if (!strongWrapper) {
+    return;
+  }
+
```

Comment:
I think these checks have no purpose here, because `buffer` is created on this thread (this is audio tap thread) and it always exist here.

I added these checks to the JS thread, because I believe there may be some cases, when that JS-thread code gets called too late and something bad happens to the buffer and is no longer available. 

This is the case that @wkozyra95 commented above. I think we can either copy the buffer and pass it to the JS thread or just skip if it's no longer available (like I do now).